core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -52,9 +61,73 @@ public SigningStage(HttpClientDependencies dependencies) {
     @Override
     public SdkHttpFullRequest execute(SdkHttpFullRequest request, RequestExecutionContext context) throws Exception {
         InterruptMonitor.checkInterrupted();
+        if (context.executionAttributes().getAttribute(SdkExecutionAttribute.SELECTED_AUTH_SCHEME) != null) {
+            return sraSignRequest(request, context);
+        }
         return signRequest(request, context);
     }
 
+    private <T extends Identity> SdkHttpFullRequest sraSignRequest(SdkHttpFullRequest request, RequestExecutionContext context) {
+
+        ExecutionAttributes executionAttributes = context.executionAttributes();
+        SelectedAuthScheme<T> selectedAuthScheme =
+            executionAttributes.getAttribute(SdkExecutionAttribute.SELECTED_AUTH_SCHEME);
+
+        AuthSchemeOption authSchemeOption = selectedAuthScheme.authSchemeOption();
+
+        // TODO: Identity resolution should move to before Endpoint resolution interceptor, to support accountId based endpoints
+        //  and also to logically separate out identity resolution as its own step.
+        ResolveIdentityRequest.Builder identityRequestBuilder = ResolveIdentityRequest.builder();
+        authSchemeOption.forEachIdentityProperty(identityRequestBuilder::putProperty);
+
+        IdentityProvider<T> identityProvider = selectedAuthScheme.identityProvider();
+        T identity = identityProvider.resolveIdentity(identityRequestBuilder.build()).join();
+
+        // TODO: resolveIdentity(Consumer builder) doesn't get used
+
+        // Consumer<ResolveIdentityRequest.Builder> c = r -> r.putProperty(IdentityProperty.create(String.class, "key"), "value");
+        // identityProvider.resolveIdentity(c);
+        // TODO: Consumer Builder is not working because of ambiguity!!
+        // identityProvider.resolveIdentity(r -> r.putProperty(IdentityProperty.create(String.class, "key"), "value"));
```

Comment:
I think that's because ResolveIdentityRequest has only one abstract method - `putIdentityProperty`. I learnt in a different [PR](https://github.com/aws/aws-sdk-java-v2/pull/4100/files#r1230091719) that making this ToCopyableBuilder will make it not a functional interface and I think making ResolveIdentityRequest copyable sounds good, so I can do that in separate PR. LMK thoughts.